sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql

```diff
@@ -0,0 +1,130 @@
+WITH events_unnested AS (
+  SELECT
+    DATE_TRUNC(submission_timestamp, DAY) AS submission_date,
+    category AS event_category,
+    name AS event_name,
+    timestamp AS event_timestamp,
+    client_info,
+    metadata,
+    normalized_os,
+    normalized_os_version,
+    normalized_country_code,
+    normalized_channel,
+    ping_info,
+    extra AS event_details,
+    metrics
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category IN ('newtab', 'topsites', 'newtab.search', 'newtab.search.ad')
+    AND name IN ('closed', 'opened', 'impression', 'issued', 'click')
+),
+categorized_events AS (
+  SELECT
+        -- Unique Identifiers
+    client_info.client_id,
+    mozfun.map.get_key(event_details, "newtab_visit_id") AS newtab_visit_id,
+        -- Metrics
+    event_name = "issued"
+    AND event_category = "newtab.search" AS is_search_issued,
+        -- ??? is_tagged_search
+        -- ??? is_follow_on_search
+    event_name = "impression"
+    AND event_category = 'newtab.search.ad'
+    AND mozfun.map.get_key(event_details, "is_tagged") = "true" AS is_tagged_search_ad_impression,
+    event_name = "impression"
+    AND event_category = 'newtab.search.ad'
+    AND mozfun.map.get_key(
+      event_details,
+      "is_follow_on"
+    ) = "true" AS is_follow_on_search_ad_impression,
+    event_name = "click"
+    AND event_category = 'newtab.search.ad'
+    AND mozfun.map.get_key(event_details, "is_tagged") = "true" AS is_tagged_search_ad_click,
+    event_name = "click"
+    AND event_category = 'newtab.search.ad'
+    AND mozfun.map.get_key(event_details, "is_follow_on") = "true" AS is_follow_on_search_ad_click,
+    event_category = 'topsites'
+    AND event_name = 'impression' AS is_topsite_impression,
+    event_category = 'topsites'
+    AND event_name = 'impression'
+    AND mozfun.map.get_key(
+      event_details,
+      "is_sponsored"
+    ) = "true" AS is_sponsored_topsite_impression,
+    event_category = 'topsites'
+    AND event_name = 'click' AS is_topsite_click,
+    event_category = 'topsites'
+    AND event_name = 'click'
+    AND mozfun.map.get_key(event_details, "is_sponsored") = "true" AS is_sponsored_topsite_click,
+    IF(event_name = "opened", event_timestamp, NULL) AS newtab_visit_started_at,
+    IF(event_name = "closed", event_timestamp, NULL) AS newtab_visit_ended_at,
+        -- Client/Session-unique attributes
+    normalized_os,
+    normalized_os_version,
+    normalized_country_code,
+    normalized_channel,
+    client_info.app_display_version,
+    mozfun.map.get_key(event_details, "source") AS newtab_open_source,
+    metrics.string.search_engine_private_engine_id AS default_search_engine,
+    metrics.string.search_engine_default_engine_id AS default_private_search_engine,
+    ping_info.experiments,
+        -- ??? private_browsing_mode
```

Comment:
what are the `???`?

Reply:
These are meant to be added in a future PR (upstream data changes are required)